qt/widgets/common/inc/MantidQtWidgets/Common/WorkspaceMultiSelector.h

```diff
@@ -0,0 +1,125 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2011 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllOption.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include <QStringList>
+#include <QTableWidget>
+#include <mutex>
+
+#include <Poco/AutoPtr.h>
+#include <Poco/NObserver.h>
+
+typedef std::vector<std::pair<std::string, std::string>> stringPairVec;
+namespace MantidQt {
+namespace MantidWidgets {
+/**
+This class defines a widget for selecting multiple workspace present in the
+AnalysisDataService and adding to a QTableWidget.
+
+Subscribes to the WorkspaceAddNotification and WorkspaceDeleteNotification from
+the ADS.
+
+Types of workspace to show can be restricted in several ways:
+
+ * By listing allowed WorkspaceIDs to show (Workspace2D, TableWorkspace, etc)
+ * By deciding whether or not to show "hidden" workspaces (identified with a
+double underscore at the start of the
+          workspace name
+ * By providing a suffix that the workspace name must have
+*/
+class EXPORT_OPT_MANTIDQT_COMMON WorkspaceMultiSelector : public QTableWidget {
```

Comment:
Ah I've just spotted this class shares some code with `WorkspaceSelector`. Would it be possible to re-use some of the common code in some way so we can avoid duplicates?

Maybe they could both inherit from a class which holds the common functionality?

Reply:
As discussed this morning, we may investigate in a separate issue whether this is something we want to do.